crates/turborepo-cache/src/signature_authentication.rs

```diff
@@ -0,0 +1,271 @@
+use std::env;
+
+use base64::{prelude::BASE64_STANDARD, Engine};
+use os_str_bytes::OsStringBytes;
+use ring::{
+    hmac,
+    hmac::{Algorithm, Tag, HMAC_SHA256},
+};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum SignatureError {
+    #[error(
+        "signature secret key not found. You must specify a secret key in the \
+         TURBO_REMOTE_CACHE_SIGNATURE_KEY environment variable"
+    )]
+    NoSignatureSecretKey,
+    #[error("serialization error: {0}")]
+    SerializationError(#[from] serde_json::Error),
+    #[error("base64 encoding error: {0}")]
+    Base64EncodingError(#[from] base64::DecodeError),
+}
+
+static TURBO_HMAC_ALGORITHM: Algorithm = HMAC_SHA256;
```

Comment:
this is static because apparently you can't have a const that refers to another const